### ISSUE TITLE
Sphinx conf.py needs_version entry is outdated

### ISSUE THREAD

**Issue by 251967bd-8758-482b-b183-bab71c7635ec** [python/cpython#86986]
BPO | [42820](https://bugs.python.org/issue42820)
--- | :---
Nosy | @terryjreedy, @JulienPalard, @timhoffm

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-01-04.01:28:14.798>
labels = ['docs']
title = 'Sphinx conf.py needs_version entry is outdated'
updated_at = <Date 2021-01-08.23:35:42.562>
user = 'https://github.com/timhoffm'
```

bugs.python.org fields:
```python
activity = <Date 2021-01-08.23:35:42.562>
actor = 'terry.reedy'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2021-01-04.01:28:14.798>
creator = 'timhoffm'
dependencies = []
files = []
hgrepos = []
issue_num = 42820
keywords = []
message_count = 2.0
messages = ['384305', '384697']
nosy_count = 4.0
nosy_names = ['terry.reedy', 'docs@python', 'mdk', 'timhoffm']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue42820'
versions = []
```

</p></details>

**Comment by JulienPalard** [comment]
This is now related to https://github.com/python/cpython/pull/92318, the new needs_sphinx is 3.2, cc @sorcio.

**Comment by ambv** [comment]
Fixed on 3.10, 3.11, and 3.12.
